apps/angular/projection/src/app/data-access/student.store.ts

```diff
@@ -1,23 +1,21 @@
-import { Injectable } from '@angular/core';
-import { BehaviorSubject } from 'rxjs';
+import { Injectable, signal } from '@angular/core';
 import { Student } from '../model/student.model';
 
 @Injectable({
   providedIn: 'root',
 })
 export class StudentStore {
-  private students = new BehaviorSubject<Student[]>([]);
-  students$ = this.students.asObservable();
+  students = signal<Student[]>([]);
 
   addAll(students: Student[]) {
-    this.students.next(students);
+    this.students.set(students);
   }
 
   addOne(student: Student) {
-    this.students.next([...this.students.value, student]);
+    this.students.set([...this.students(), student]);
```

Comment:
Why not use `update` here?

Reply:
you can use "update" too to get previous value and add new student, but since u current have the students list, u can also use the set() method